iree/turbine/kernel/wave/workgroup_reordering.py

```diff
@@ -0,0 +1,34 @@
+from .._support.tracing import CapturedTrace
+from .._support.indexing import *
+from ..ops.wave_ops import *
+from ..lang.global_symbols import *
+from .constraints import *
+from .utils.symbol_utils import *
+
+
+def reorder_workgroups(graph: CapturedTrace, reordering_constraints):
+    new_wg0, new_wg1, new_wg2 = None, None, None
+    for c in reordering_constraints:
+        if c.wg_dim == WORKGROUP_0:
+            new_wg0 = c.reordered_equation
+        elif c.wg_dim == WORKGROUP_1:
+            new_wg1 = c.reordered_equation
+        else:
+            new_wg2 = c.reordered_equation
+
+    if new_wg0 is not None or new_wg1 is not None or new_wg2 is not None:
```

Comment:
NIT: Wave/IREE/MLIR style programming, do early exit instead of creating more nested regions. i.e If no new_wg is found, just return